lib/galaxy/jobs/__init__.py

```diff
@@ -1492,7 +1499,8 @@ def path_rewriter( path ):
 
         # fix permissions
         for path in [ dp.real_path for dp in self.get_mutable_output_fnames() ]:
-            util.umask_fix_perms( path, self.app.config.umask, 0o666, self.app.config.gid )
+            if os.path.exists(path):
+                util.umask_fix_perms( path, self.app.config.umask, 0o666, self.app.config.gid )
```

Comment:
That method does its own logging though - at ``log.exception`` it assumes the path exists and this piece of code shouldn't assume the path exists. I feel like this check is the correct thing to do.